src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -9,16 +9,28 @@
     [TestFixture]
     public class MsmqQueueCreatorTests
     {
+
+        [TearDown]
+        public void TearDown()
+        {
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.receiver");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.target1");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.target2");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.txreceiver");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.MsmqDefaultPermissions");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.NoChangesToExisting");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.permissions");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.too.long.B2568312E4BF424D8C875D0CD3F65A3E");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.badidentity");
+            DeleteQueueIfPresent(GetReallyLongQueueName());
```

Comment:
I definitely don't like approach that would require you to add every used queue to the `TearDown` method additionally. If we go with a `TearDown` approach, some more convention based approach would be appropriate (e.g. delete all queues starting with `MsmqQueueCreatorTests`). Otherwise I think deleting them in a finally block within the test is more expressive.